### PR TITLE
Golang instrumentation featuregates into cli

### PR DESCRIPTION
**Pull request by yuriolisa**
**Description:** <Describe what has changed.>

Changed featuregates.autoinstrumentation.go to cli. 
**Link to tracking Issue(s):** <Issue number if applicable>

- Resolves: #2675 

**Testing:** <Describe what testing was performed and which tests were added.>

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -136,6 +137,7 @@ func main() {
 	pflag.BoolVar(&enableMultiInstrumentation, "enable-multi-instrumentation", false, "Controls whether the operator supports multi instrumentation")
 	pflag.BoolVar(&enableApacheHttpdInstrumentation, constants.FlagApacheHttpd, true, "Controls whether the operator supports Apache HTTPD auto-instrumentation")
 	pflag.BoolVar(&enableDotNetInstrumentation, constants.FlagDotNet, true, "Controls whether the operator supports dotnet auto-instrumentation")
+	pflag.BoolVar(&enableGoInstrumentation, constants.FlagGo, true, "Controls whether the operator supports Go auto-instrumentation")
```
Comment (TylerHelmuth) [inline main.go:140]:
```suggestion
	pflag.BoolVar(&enableGoInstrumentation, constants.FlagGo, false, "Controls whether the operator supports Go auto-instrumentation")
```

This instrumentation should be disabled by default

File: .chloggen/featuregates-into-cli.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: 'breaking'
+
+# The name of the component, or a single word describing the area of concern, (e.g. collector, target allocator, auto-instrumentation, opamp, github action)
+component: 'operator'
+
+# A brief description of the change. Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Change featuregates auto-instrumentation.go to CLI
```
Comment (pavolloffay) [inline .chloggen/featuregates-into-cli.yaml:8]: we should document exactly which flag is changing and what is the new flag.

**Review by pavolloffay — COMMENTED** [review]
LGTM.

the e2e tests are failing, if you re-push it should pass

**Comment by swiatekm-sumo** [comment]
The reason these tests fail is that this feature gate is enabled in the default manifests. #1555 added this, and I'm not entirely sure why. @TylerHelmuth do you recall?

**Comment by TylerHelmuth** [comment]
@swiatekm-sumo back then there was no other way (that I knew of at least) to enable the feature for the tests.  Specific discussion here: https://github.com/open-telemetry/opentelemetry-operator/pull/1555#discussion_r1160122613

**Comment by swiatekm-sumo** [comment]
> @swiatekm-sumo back then there was no other way (that I knew of at least) to enable the feature for the tests. Specific discussion here: [#1555 (comment)](https://github.com/open-telemetry/opentelemetry-operator/pull/1555#discussion_r1160122613)

Is there actually any difference between enabling the flag by default in the code vs doing so in the manifests? I want to understand if this instrumentation should be enabled or disabled by default. Right now, we have the ability to add operator flags for specific E2E test suites.

**Comment by TylerHelmuth** [comment]
> Right now, we have the ability to add operator flags for specific E2E test suites.

Sweet, we should enable it for the tests that need it, but I think disabled by default is still correct since it is alpha.

**Comment by TylerHelmuth** [comment]
Clicked the wrong button

**Comment by swiatekm-sumo** [comment]
@yuriolisa it sounds like you need to add something the following:

```
.PHONY: prepare-e2e-with-operator-flags
prepare-e2e-with-operator-flags: OPERATOR_ARG = $(OPERATOR_FLAGS)
prepare-e2e-with-operator-flags chainsaw  add-operator-arg prepare-e2e
```

And then use it in the CI for setup of instrumentation tests.

**Review by swiatekm-sumo — APPROVED** [review]

**Review by pavolloffay — APPROVED** [review]

**Pull request merged**